### PR TITLE
Added GTM data attributes

### PR DESCRIPTION
**Pull request by justinvandenberg**
Sorry about all the build files. Pantheon makes us track them. 

Changes:
* Added `data-track-gtm` attributes
* Added `dataLayer` pushes to audio and video components
* Changed `oembed` to `text` fields

Time tracking:
[NPM] -> [01] Redesign
If needed, Grace can add you.

### CODE REVIEW

File: web/themes/npm/assets/scripts/modules/audio.js [hunk 1]
```diff
@@ -73,21 +73,22 @@ export default class Audio {
     }
 
     this.skipBackwardTrigger.addEventListener('click', this.skipBackward)
-    this.wavesurfer.on('audioprocess', this.audioProcess)
+    this.wavesurfer.on('audioprocess', this.audioProgress)
```
Comment (olets) [inline web/themes/npm/assets/scripts/modules/audio.js]: just making sure: the event should be `audioprocess`?

File: web/themes/npm/assets/scripts/modules/audio.js [hunk 2]
```diff
@@ -97,4 +98,14 @@ export default class Audio {
...
     e.preventDefault()
   }
+
+  pushToDataLayer = () => {
+    window.dataLayer = window.dataLayer || []
+    dataLayer.push({
+      event: 'dataLayer push event',
+      event_category: 'Audio',
+      event_action: this.player.dataset.audioTitle || 'Audio Title',
+      event_label: 'Play',
```
Comment (olets) [inline web/themes/npm/assets/scripts/modules/audio.js]: making sure: okay to have pause events labeled `Play`?

File: web/themes/npm/assets/scripts/modules/video.js [hunk 3]
```diff
@@ -18,8 +18,20 @@ export default class Video {
...
     e.preventDefault()
   }
+
+  pushToDataLayer = () => {
+    window.dataLayer = window.dataLayer || []
+    dataLayer.push({
+      event: 'dataLayer push event',
+      event_category: 'Video',
+      event_action: this.player.dataset.videoTitle || 'Video Title',
+      event_label: 'Play',
```
Comment (olets) [inline web/themes/npm/assets/scripts/modules/video.js]: same pause question here

File: web/themes/npm/views/_blocks/articles/article-large.twig [hunk 4]
```diff
@@ -28,8 +28,15 @@
...
-      {{ article.button_text | length ? article.button_text : 'Read more' }}
+    {% set button_link = article.button_link | length ? article.button_link : article.link %}
+    {% set button_text = article.button_text | length ? article.button_text : 'Read more' %}
+
+    <a
+      class="arrow mt-20 text-14"
+      href="{{ button_link }}"
+      data-track-gtm="Call to Action|{{ current_url }}|{{ button_text }}:{{ button_link }}"
+    >
+      {{ button_text }}
```
Comment (olets) [inline web/themes/npm/views/_blocks/articles/article-large.twig]: nice refactor 👍

File: web/themes/npm/views/_blocks/featured-article/index.twig [hunk 5]
```diff
@@ -27,8 +27,16 @@
...
 
-          <a class="button mt-25 lg:mt-30" href="{{ article.button_link is not empty ? article.button_link : article.link }}">
-            {{ article.button_text is not empty ? article.button_text : 'Read more' }}
+          {% set button_link = article.button_link | length ? article.button_link : article.link %}
+          {% set button_text = article.button_text | length ? article.button_text : 'Read more' %}
+
+          <a
+            class="button mt-25 lg:mt-30"
+            href="{{ button_link }}"
+            data-track-gtm="Call to Action|{{ current_url }}|{{ button_text }}:{{ button_link }}"
```
Comment (olets) [inline web/themes/npm/views/_blocks/featured-article/index.twig]: not a blocker at all, could be cool to have a macro that `|join()`s the elements in an array and gives you the whole data attribute

File: web/themes/npm/views/_blocks/featured-report/index.twig [hunk 6]
```diff
@@ -17,8 +17,15 @@
...
-            {{ report.button_text is not empty ? report.button_text : 'Read more' }}
+          {% set button_link =  report.button_link | length ? report.button_link : report.link %}
+          {% set button_text =  report.button_text | length ? report.button_text : 'Read more' %}
+
+          <a
+            class="button-outline-white mt-25 lg:mt-30"
+            href="{{ button_link }}"
+            data-track-gtm="Call to Action|{{ current_url }}|{{ button_text }}:{{ button_link }}"
+          >
+            {{ button_text }}
```
Comment (olets) [inline web/themes/npm/views/_blocks/featured-report/index.twig]: seen a few of these… could the whole link be a macro/partial/what have you? (shouldn't block this pr)

File: web/themes/npm/views/_partials/video/index.twig [hunk 7]
```diff
@@ -1,23 +1,25 @@
-{% set video = video ?? null %}
+{% set video = video ? fn('npm_get_oembed_details', video) : null %}
```
Comment (olets) [inline web/themes/npm/views/_partials/video/index.twig]: is this wordpress sorcery?

File: web/themes/npm/assets/scripts/modules/audio.js (hunk 2 above)
Comment (justinvandenberg) [reply web/themes/npm/assets/scripts/modules/audio.js]: Good callout, I'll ask Aakash if he wants that or even pause events at all.

File: web/themes/npm/assets/scripts/modules/audio.js (hunk 1 above)
Comment (justinvandenberg) [reply web/themes/npm/assets/scripts/modules/audio.js]: Yeah, that's what the plugin uses, but it seems to make more sense labeled as progress.

File: web/themes/npm/assets/scripts/modules/video.js (hunk 3 above)
Comment (justinvandenberg) [reply web/themes/npm/assets/scripts/modules/video.js]: This one should only fire on the initial play. Maybe I should do the same for the other?

File: web/themes/npm/views/_blocks/featured-article/index.twig (hunk 5 above)
Comment (justinvandenberg) [reply web/themes/npm/views/_blocks/featured-article/index.twig]: Good thinking. If I can sneak in some extra time I'll add that.

File: web/themes/npm/views/_partials/video/index.twig (hunk 7 above)
Comment (justinvandenberg) [reply web/themes/npm/views/_partials/video/index.twig]: Timber calling a function. First arg is the function followed by each actual arg. It's weird.

Comment (olets) [reply web/themes/npm/views/_partials/video/index.twig]: makes sense, thanks!

**Pull request merged**